src/com/moclojer/rq/queue.clj

```diff
@@ -69,5 +71,252 @@
+  
+  Parameters:
+  - client: Redis client
+  - queue-name: Name of the queue
+  - index: specific index to access"
+  [client queue-name index & options]
+  (let [{:keys [pattern]
+         :or {pattern :rq}} options
+        packed-queue-name (utils/pack-pattern pattern queue-name)
+        return (.lindex @client packed-queue-name index)]
+
+    (let [message (clojure.edn/read-string return)]
+      (log/debug "message found"
+                 {:client client
+                  :queue-name packed-queue-name
+                  :index index
+                  :message message})
+      message)))
+
+(defn lset
+  "Set a new message in a specified index
+
+  Parameters:
+  - client: Redis client
+  - queue-name: Name of the queue
+  - index: specific index to access
+  - message: new msg to be added"
+  [client queue-name index message & options]
+  (let [{:keys [pattern]
+         :or {pattern :rq} :as opts} options
+        packed-queue-name (utils/pack-pattern pattern queue-name)
+        encoded-message (str (clojure.edn/read-string (str message)))
+        return (.lset @client packed-queue-name index encoded-message)]
+
+    (log/debug "set in queue"
+               {:client client
+                :queue-name packed-queue-name
+                :message (str encoded-message)
+                :index index
+                :options opts
+                :return return})
+    return))
+
+(defn lrem
+  "Removes a specified occurance of the message given (if any)
+  
+  Parameters:
+  - client: Redis client
+  - queue-name: Name of the queue
+  - msg: new msg to be added
+  - cnt: count
+    count > 0: Remove elements equal to element moving from head to tail.
+    count < 0: Remove elements equal to element moving from tail to head.
+    count = 0: Remove all elements equal to element."
+  [client queue-name cnt msg & options]
+  (let [{:keys [pattern]
+         :or {pattern :rq}} options
+        packed-queue-name (utils/pack-pattern pattern queue-name)
+        return (.lrem @client packed-queue-name cnt (str msg))]
+
+    (log/debug "removed from queue"
+               {:client client
+                :queue-name queue-name
+                :msg (clojure.edn/read-string (str msg))
+                :count cnt
+                :return return})
+    return))
+
+(defn linsert
+  "insert a message before the first occurance of a pivot given.
+
+  parameters:
+  - client: redis client
+  - queue-name: name of the queue
+  - msg: new msg to be added
+  - pivot: pivot message to be added before or after
+  - options:
+    - pos (keywords):
+      - before: insert the message before the pivot
+      - after: insert the message after the pivot"
+  [client queue-name pivot msg & options]
+  (let [{:keys [pos pattern]
+         :or {pos :before
+              pattern :rq} :as opts} options
+        packed-queue-name (utils/pack-pattern pattern queue-name)
+        encoded-message (str (clojure.edn/read-string (str msg)))
```

Comment:
@Felipe-gsilva This does the same operation thrice. Parse a clojure seq to string, them read that string back to a clojure seq, and then parse it back to a string??

It's better to just use clj's builtin `pr-str` helper. Anyways, I won't request any changes (if I did, this would have to be improved all over the changes) because the implementation won't be used as soon as we merge the dynamic fns queue.

Reply:
Great!